core/templates/dev/head/components/SearchBarDirective.js

```diff
@@ -110,17 +110,28 @@ oppia.directive('searchBar', [function() {
         };
 
         var _searchBarFullyLoaded = false;
+        var url;
 
         var _hasChangedSearchQuery = Boolean(urlService.getUrlParams().q);
-        var _onSearchQueryChangeExec = function() {
-          searchService.executeSearchQuery(
+        var _onSearchQueryChangeExec = function(reloadPage) {
+          $scope.searchIsLoading = true;
+          url = searchService.executeSearchQuery(
```

Comment:
This is a bit weird semantically -- I'm not sure that sending the search query to the backend and composing the frontend URL are related. Can the url be computed in a separate, independent, call?